### ISSUE TITLE
NoneType has no len() error. 

### ISSUE THREAD

**Issue by joshuaLW** [fishtown-analytics/dbt#3014]
### Describe the bug
A clear and concise description of what the bug is. What command did you run? What happened?

### Steps To Reproduce
In as much detail as possible, please provide steps to reproduce the issue. Sample data that triggers the issue, example model code, etc is all very helpful here.

### Screenshots and log output
If applicable, add screenshots or log output to help explain your problem.

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
<output goes here>
```

**The operating system you're using:**

**The output of `python --version`:**

**Comment by jtcohen6** [comment]
Hey @joshuaLW, this is indeed a cryptic error. The question of removing the Snowplow package seems like a good place to start debugging.

- Do you have anything else in `packages.yml`?
- Do you still have references to the `snowplow` package in `dbt_project.yml`?

**Comment by joshuaLW** [comment]
@jtcohen6  three things... I never actually checked that the snowplow modules were removed from packages.yml when i ran "dbt clean" but now having reversed these changes with "dbt deps" I've only this in that file:
packages:
  - package: fishtown-analytics/snowplow
    version: 0.11.0

I did comment out all variables and models referencing the snowplow package in the project.yml

I didn't remove all dependencies upon snowplow from my models  but i expected to get an error of the kind model x refers to model y that does not exist. 

I'm going to go at it that way now actually: i'm going to remove references to snowplow models in my models first and then try removing snowplow module again.

**Comment by jtcohen6** [comment]
> I did comment out all variables and models referencing the snowplow package in the project.yml

If I had to pick one of the three as the likeliest cause, I'd say it's this one. Try removing `snowplow` from `packages.yml`, run `dbt clean`, leave the `vars` in place, and then see what error you get back. (I hope it will be of the missing-ref kind.)

**Comment by joshuaLW** [comment]
@jtcohen6  good shout.  I've just tried emptying the `packages.yml` then running `dbt clean` and i'm getting the errors i expected  eg. "model x depends on y" so i'll keep going with cleaning up my models till all dependancies are removed... and hopefully it was just that... :) FWIW I thought dbt clean would have cleaned my packages.yml too... but computers can't read my mind can they?

**Comment by joshuaLW** [comment]
@jtcohen6  yup  i found the line that was causing the problem.  It was a reference to a variable that was nested in the package i'd removed. 
![image](https://user-images.githubusercontent.com/71407752/104730946-db0d5f00-5732-11eb-9651-4da2a340a584.png)

**Comment by jtcohen6** [comment]
There it is! Well found.

I think it would be an improvement to the snowplow package if its models handled better the case when `var('snowplow:pass_through_columns')` is undefined in the project (`None`), rather than merely empty (`[]`).

Perhaps by rewriting that line to be:
```
{% if var('snowplow_pass_through_columns', []) | length > 0 %}
```

That way, if you do want to copy-paste the model code into your own project, and (sensibly) remove the snowplow-specific `vars` from your project config, you won't be stymied by a cryptic error.

I'm going to transfer this issue to the snowplow package repository, since that's where the change should take place.
